src/main/scala/sbtprotoc/LoggingOutputStream.scala

```diff
@@ -0,0 +1,37 @@
+package sbtprotoc
+
+import java.io.{ByteArrayOutputStream, OutputStream}
+
+import sbt.{Level, Logger}
+
+/** Split an OutputStream into messages and feed them to a given logger at a specified level. Not thread-safe. */
+class LoggingOutputStream(logger: Logger, level: Level.Value, separator: String)
```

Comment:
Is it possible that SBT provides a solution for this use case? I am concerned that this is brittle. For example, if I read the code correctly, I think that if the last line doesn't terminate with a new line, it will never be logged.

Reply:
In addition, if there's nothing we can use, maybe it would be better to separate concerns and have the `LoggingOutputStream` call a callback `String=>Unit` for each line instead of being couple to a `Logger`. In that case its name should be something like `LineBasedOutputStream`.

Reply:
https://github.com/sbt/util/blob/d31b9c509384ecaf37f2ffaec8731250e3c60532/internal/util-logging/src/main/scala/sbt/internal/util/LoggerWriter.scala has similar goals, but is "internal" and can't be used as-is as it's not `OutputStream`-centric.

It does handle trailing, non-terminated lines via `close()`, which is something I can/ should port indeed.

As for separating concerns, good idea. I'd be more inclined to use `message` (also used in the logger signature) over `line` though, since the separator does not need to be a line break, but I can't find anything better than `MessageSplitterOutputStream`.